core/src/main/java/org/radargun/stages/test/legacy/LegacyStressor.java

```diff
@@ -87,15 +91,18 @@ private void runInternal() {
                if (ongoingTx != null) {
                   endTransactionAndRegisterStats(null);
                }
-               stats.begin();
-               this.started = started;
                break;
-            }
-            Operation operation = operationSelector.next(random);
-            try {
-               logic.run(operation);
-            } catch (OperationLogic.RequestException e) {
-               // the exception was already logged in makeRequest
+            } else {
```

Comment:
Because when one thread starts a request, there should be already some other overlapping request being executed - otherwise the request would be uncontended (and likely faster).